ansible_collections/arista/avd/roles/eos_validate_state/python_modules/tests/avdtestconnectivity.py

```diff
@@ -33,50 +29,36 @@ def test_definition(self) -> dict | None:
+        for idx, interface in enumerate(ethernet_interfaces):
+            if not self.validate_data(data=interface, data_path=f"ethernet_interfaces.[{idx}]", required_keys=required_keys, type="routed", shutdown=False):
+                continue
 
-                peer_ethernet_interfaces = get(self.hostvars, f"{ethernet_interface['peer']}.ethernet_interfaces", required=True)
-                peer_interface = get_item(
-                    peer_ethernet_interfaces,
-                    "name",
-                    ethernet_interface["peer_interface"],
-                    required=True,
-                    var_name=f"name: {ethernet_interface['peer_interface']}",
-                )
-                peer_interface_ip = get(peer_interface, "ip_address", required=True)
-
-                if ethernet_interface["type"] == "routed":
-                    src_ip = str(ip_interface(ethernet_interface["ip_address"]).ip)
-                    dst_ip = str(ip_interface(peer_interface_ip).ip)
-                    custom_field = (
-                        f"Source: {self.device_name}_{ethernet_interface['name']} - Destination:"
-                        f" {ethernet_interface['peer']}_{ethernet_interface['peer_interface']}"
-                    )
-                    anta_tests.append(
-                        {
-                            "VerifyReachability": {
-                                "hosts": [{"source": src_ip, "destination": dst_ip, "vrf": "default", "repeat": 1}],
-                                "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": custom_field},
-                            }
-                        }
-                    )
+            if not self.is_peer_available(peer := interface["peer"]):
+                continue
 
-            except AristaAvdMissingVariableError as e:
-                LOGGER.info("Variable '%s' is missing. Please validate the Ethernet interfaces data model of this host and his peer(s).", str(e))
+            if (
+                peer_interface_ip := self.get_interface_ip(interface_model="ethernet_interfaces", interface_name=interface["peer_interface"], host=peer)
+            ) is None:
                 continue
 
-        return {self.anta_module: anta_tests}
+            src_ip = str(ip_interface(interface["ip_address"]).ip)
+            dst_ip = str(ip_interface(peer_interface_ip).ip)
+            custom_field = f"Source: {self.device_name}_{interface['name']} - Destination: {peer}_{interface['peer_interface']}"
+            anta_tests.append(
+                {
+                    "VerifyReachability": {
+                        "hosts": [{"source": src_ip, "destination": dst_ip, "vrf": "default", "repeat": 1}],
+                        "result_overwrite": {"categories": self.categories, "description": self.description, "custom_field": custom_field},
+                    }
+                }
+            )
```

Comment:
not for this PR but I think we should find a way to make this look nicer at some point